python/lsst/obs/hsc/hscPupil.py

```diff
@@ -0,0 +1,137 @@
+# LSST Project (http://www.lsst.org/).
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the LSST License Statement and
+# the GNU General Public License along with this program.  If not,
+# see <http://www.lsstcorp.org/LegalNotices/>.
+#
+
+from lsst.afw.cameraGeom import PupilFactory
+from lsst.afw.geom import Angle, degrees
+import numpy as np
+
+
+class HscPupilFactory(PupilFactory):
+    """!Pupil obscuration function factory for HSC.
+    """
+    def __init__(self, *args, **kwargs):
+        """!Construct a PupilFactory.
+
+        @param[in] visitInfo  VisitInfo object for a particular exposure.
+        @param[in] pupilSize  Size in meters of constructed Pupils.
+        @param[in] npix       Constructed Pupils will be npix x npix.
+        """
+        super(HscPupilFactory, self).__init__(*args, **kwargs)
+
+        hra = self._horizonRotAngle()
+        hraRad = hra.asRadians()
+        rot = np.array([[ np.cos(hraRad), np.sin(hraRad)],
+                        [-np.sin(hraRad), np.cos(hraRad)]])
+
+        # Compute spider shadow parameters accounting for rotation angle.
+        # Location where pairs of struts meet near prime focus.
+        unrotStartpos = [np.array([0.43, 0.43]),
+                         np.array([0.43, 0.43]),
+                         np.array([-0.43, -0.43]),
+                         np.array([-0.43, -0.43])]
+        # Half angle between pair of struts that meet at Subaru prime focus
+        # ring.
+        strutAngle = 51.75*degrees
+        alpha = strutAngle - 45.0*degrees
+        unrotAngles = [90*degrees + alpha,
+                       -alpha,
+                       180*degrees - alpha,
+                       270*degrees + alpha]
+        # Apply rotation and save the results
+        self._spiderStartpos = []
+        self._spiderAngles = []
+        for pos, angle in zip(unrotStartpos, unrotAngles):
+            self._spiderStartpos.append(np.dot(rot, pos))
+            self._spiderAngles.append(angle - hra)
+
+    def _horizonRotAngle(self):
+        """!Compute rotation angle of camera with respect to horizontal
+        coordinates from self.VisitInfo.
+
+        @returns horizon rotation angle.
+        """
+        observatory = self.visitInfo.getObservatory()
+        lat = observatory.getLatitude()
+        lon = observatory.getLongitude()
+        radec = self.visitInfo.getBoresightRaDec()
+        ra = radec.getRa()
+        dec = radec.getDec()
+        era = self.visitInfo.getEra()
```

Comment:
Thanks for the notice.  I think approximate is good enough for now, but I'll keep this in mind going forward.